pil-analyzer/src/expression_processor.rs

```diff
@@ -196,10 +196,13 @@ impl<'a, D: AnalysisDriver> ExpressionProcessor<'a, D> {
                     .map(|args| args.into_iter().map(|t| self.process_type(t)).collect());
 
                 Expression::StructExpression(
-                    src,
+                    src.clone(),
                     StructExpression {
                         name: Reference::Poly(PolynomialReference {
-                            name: self.driver.resolve_ref(&name.path, SymbolCategory::Struct),
+                            name: self
+                                .driver
+                                .resolve_ref(&name.path, SymbolCategory::Struct)
+                                .unwrap(),
```

Comment:
I'm still quite uneasy about this because it is difficult to search for. Can you maybe use `expect` with something like `"TODO pass this up"`?